secret.go

```diff
@@ -58,30 +57,35 @@ type Secret struct {
 }
 
 // ModeValue function helps by converting a textual mode to the expected value for fuse.
-func (s Secret) ModeValue() os.FileMode {
+func (s Secret) ModeValue() (os.FileMode, error) {
 	mode := s.Mode
 	if mode == "" {
 		mode = "0440"
 	}
 	modeValue, err := strconv.ParseUint(mode, 8 /* base */, 16 /* bits */)
 	if err != nil {
-		log.Printf("Unable to convert secret mode (%v) to octal, using '0440': %v\n", mode, err)
-		modeValue = 0440
+		return 0, fmt.Errorf("Unable to parse secret mode (%v): %v\n", mode, err)
```

Comment:
I would maybe add the word "file" somewhere in here, to note that it is a file mode.